tests/emtests/_common.rs

```diff
@@ -0,0 +1,110 @@
+use std::env;
+use wasmer_runtime::Backend;
+
+pub fn get_backend() -> Option<Backend> {
+    #[cfg(feature = "backend-cranelift")]
+    {
+        if let Ok(v) = env::var("WASMER_TEST_CLIF") {
+            if v == "1" {
+                return Some(Backend::Cranelift);
+            }
+        }
+    }
+    #[cfg(feature = "backend-llvm")]
+    {
+        if let Ok(v) = env::var("WASMER_TEST_LLVM") {
+            if v == "1" {
+                return Some(Backend::LLVM);
+            }
+        }
+    }
+    #[cfg(feature = "backend-singlepass")]
+    {
+        if let Ok(v) = env::var("WASMER_TEST_SINGLEPASS") {
+            if v == "1" {
+                return Some(Backend::Singlepass);
+            }
+        }
+    }
+
+    None
+}
+
+macro_rules! assert_emscripten_output {
+    ($file:expr, $name:expr, $args:expr, $expected:expr) => {{
+
+        use wasmer_emscripten::{
+            EmscriptenGlobals,
+            generate_emscripten_env,
+        };
+        use wasmer_dev_utils::stdio::StdioCapturer;
+
+        let wasm_bytes = include_bytes!($file);
+        let backend = $crate::emtests::_common::get_backend().expect("Please set one of `WASMER_TEST_CLIF`, `WASMER_TEST_LLVM`, or `WASMER_TEST_SINGELPASS` to `1`.");
```

Comment:
Note that this error string may make a suggestion that won't help (WASM_TEST_LLVM=1 when llvm isn't compiled in).

Reply:
I'm fine with that for now, `spectests` has its own version of this code that's more robust.  Ideally we'll combine them together but that should be done in a PR where it's possible to read and understand the diff